frontend/src/metabase/embedding-sdk/theme/dynamic-css-vars-config.ts

```diff
@@ -0,0 +1,21 @@
+import type { DynamicCssVarConfig } from "../types/private/css-variables";
+
+/**
+ * These CSS variables are dynamically generated based on the theme.
+ */
+export const DYNAMIC_CSS_VARIABLES: DynamicCssVarConfig = {
+  "--mb-color-notebook-step-bg": {
+    source: "bg-white",
+    light: { darken: 0.05 },
+    dark: { lighten: 0.5 },
+  },
+  "--mb-color-notebook-step-bg-hover": {
+    source: "bg-white",
+    light: { darken: 0.1 },
+    dark: { lighten: 0.4 },
+  },
+  "--mb-color-background-hover": {
+    source: { dark: "bg-white" },
+    dark: { lighten: 0.5 },
+  },
```

Comment:
We only generate the dynamic CSS variables for SDK's dark mode in this case, and use the default background-hover for light mode. I tried to match the existing colors but I'll leave it for another PR.